ironfish/src/mining/stratum/stratumServer.ts

```diff
@@ -127,9 +132,27 @@ export class StratumServer {
     this.send(client, 'mining.wait_for_work')
   }
 
+  addToConnectionCount(remoteAddress: string, addition: number) : number {
+    this.openConnections.set(remoteAddress, (this.openConnections.get(remoteAddress) ?? 0) + addition)
+    return this.openConnections.get(remoteAddress) ?? 0
+  }
+
   private onConnection(socket: net.Socket): void {
+
     const client = StratumServerClient.accept(socket, this.nextMinerId++)
 
+    let remoteAddress = socket.remoteAddress ?? ""
+
+    if((this.openConnections.get(remoteAddress) ?? 0) + 1 > this.maxOpenConnections){
+      //Set timeout for disconnect in order to prevent spamming, then close the connection
+      setTimeout(() => {
+        client.close()
+      }, 10000)
+      return
+    }
+
+    this.addToConnectionCount(remoteAddress, 1)
```

Comment:
A few more suggestions:
1. We probably don't want to add a `remoteAddress` to the map if it isn't a falsy value (undefined, etc), otherwise we run the risk of flagging connections that are valid but not returning proper values for whatever reason. If this still ends up being an issue, we probably want to look into better ways to bucket connections.
2. Just some minor readability improvements, white space, splitting some things into vars etc.
3. Instead of doing the `+ 1 >`, we can just do `>=`
4. Adding the client to the `badClient` list so we don't send them messages also, this also prevents a possible issue where a miner connects, thinks everything is good, and then gets disconnected 10 seconds later. This would surface more dramatically, they would never fully connect and get broadcasts (although I think they'd still get the initial work/subscribed message now that I'm thinking about it - possible future improvement).

```suggestion
    const remoteAddress = socket.remoteAddress
    
    if (remoteAddress) {
      const connectionCount = this.openConnections.get(remoteAddress) ?? 0
      if (connectionCount >= this.maxOpenConnections) {
        // Add the client to the bad list so we don't waste resources sending them messages
        this.addBadClient(client)
        // Set timeout for disconnect in order to prevent spamming, then close the connection
        setTimeout(() => {
          this.clients.delete(client.id)
          client.close()
        }, 10000)
        
        return
      }
      
      this.addToConnectionCount(remoteAddress)
    }
```